backend/src/test/java/de/neuefische/backend/ChoreoIntegrationTest.java

```diff
@@ -0,0 +1,177 @@
+        Choreo choreo1 = new Choreo("1", "", List.of("1", "2", "3"));
+        choreoRepo.save(choreo1);
+        Choreo choreo2 = new Choreo("2", "", List.of("2", "3", "3"));
+        choreoRepo.save(choreo2);
+        Choreo choreo3 = new Choreo("3", "", List.of("1", "3"));
+        choreoRepo.save(choreo3);
+
+        Move move1 = new Move("1", "a", "", "", "1", "", "");
+        moveInterface.save(move1);
+        Move move2 = new Move("2", "b", "", "", "2", "", "");
+        moveInterface.save(move2);
+        Move move3 = new Move("3", "c", "", "", "3", "", "");
+        moveInterface.save(move3);
+
+        ChoreoDTO choreoDTO1 = new ChoreoDTO("1", "", List.of(move1, move2, move3));
+        ChoreoDTO choreoDTO2 = new ChoreoDTO("2", "", List.of(move2, move3, move3));
+        ChoreoDTO choreoDTO3 = new ChoreoDTO("3", "", List.of(move1, move3));
+
+        List<ChoreoDTO> expectedChoreos = List.of(choreoDTO1, choreoDTO2, choreoDTO3);
+
+        mockMvc.perform(get("/api/choreo"))
+                .andExpect(status().isOk())
+                .andExpect(content().json(objectMapper.writeValueAsString(expectedChoreos)));
+    }
+
+    @DirtiesContext
+    @Test
+    void getAllChoreos_shouldReturnEmptyList_whenNoChoreos() throws Exception {
+        mockMvc.perform(get("/api/choreo"))
+                .andExpect(status().isOk())
+                .andExpect(content().json(
+                        """
+                                []
+                                """
+                ));
+    }
+
+    @DirtiesContext
+    @Test
+    void deleteChoreo_shouldReturnReducedContent() throws Exception {
+        Choreo choreo1 = new Choreo("1", "", List.of("1", "2", "3"));
+        choreoRepo.save(choreo1);
+        Choreo choreo2 = new Choreo("2", "", List.of("2", "3", "3"));
+        choreoRepo.save(choreo2);
+        Choreo choreo3 = new Choreo("3", "", List.of("1", "3"));
+        choreoRepo.save(choreo3);
+
+        Move move1 = new Move("1", "a", "", "", "1", "", "");
+        moveInterface.save(move1);
+        Move move2 = new Move("2", "b", "", "", "2", "", "");
+        moveInterface.save(move2);
+        Move move3 = new Move("3", "c", "", "", "3", "", "");
+        moveInterface.save(move3);
+
+        ChoreoDTO choreoDTO1 = new ChoreoDTO("1", "", List.of(move1, move2, move3));
+        ChoreoDTO choreoDTO2 = new ChoreoDTO("2", "", List.of(move2, move3, move3));
+        ChoreoDTO choreoDTO3 = new ChoreoDTO("3", "", List.of(move1, move3));
+
+        String choreoToDeleteId = choreoDTO1.id();
+        List<ChoreoDTO> expectedChoreos = List.of(choreoDTO2, choreoDTO3);
+
+        mockMvc.perform(delete("/api/choreo/" + choreoToDeleteId)
+                )
+                .andExpect(status().isOk());
+
+
+        mockMvc.perform(get("/api/choreo"))
+                .andExpect(status().isOk())
+                .andExpect(content().json(objectMapper.writeValueAsString(expectedChoreos)));
+    }
```

Comment:
du könntest auch nur eine Choreo in dein Repo speichern und wenn du es löschst expectest du eine leere Liste (ist viel weniger zu schreiben und testet das gleiche, oder?)